common/src/main/java/com/wynntils/core/features/overlays/OverlayManager.java

```diff
@@ -54,13 +61,55 @@ private static void renderOverlays(RenderEvent event, OverlayInfo.RenderState re
         for (Overlay overlay : enabledOverlays) {
             OverlayInfo annotation = overlayInfoMap.get(overlay);
 
-            if (renderState != annotation.renderAt() || event.getType() != annotation.renderType()) continue;
+            if ((annotation.renderAt() == OverlayInfo.RenderState.Replace && renderState != OverlayInfo.RenderState.Pre)
```

Comment:
I suggest splitting this up in separate if statements, like:
if not the right render time - continue

if renderAt=replace:
  if not pre - continue
  setCanceled
else
 if not renderAt = state continue
fi

Or something like that.